paper/src/main/java/net/william278/huskhomes/PaperHuskHomes.java

```diff
@@ -57,4 +80,68 @@ public List<Command> registerCommands() {
                 .collect(Collectors.toList());
     }
 
+    @Override
+    public int runAsync(@NotNull Runnable runnable) {
+        if (getGlobalRegionScheduler().isPresent()) {
+            getGlobalRegionScheduler().get().run(this, scheduledTasks::add);
+            return scheduledTasks.size() - 1;
+        } else {
+            return super.runAsync(runnable);
+        }
+    }
+
+    @Override
+    public <T> CompletableFuture<T> supplyAsync(@NotNull Supplier<T> supplier) {
+        final CompletableFuture<T> future = new CompletableFuture<>();
+        if (getGlobalRegionScheduler().isPresent()) {
+            getGlobalRegionScheduler().get().run(this, scheduledTask -> future.complete(supplier.get()));
+            return future;
+        } else {
+            return super.supplyAsync(supplier);
+        }
+    }
+
+    @Override
+    public void runSync(@NotNull Runnable runnable) {
+        if (getGlobalRegionScheduler().isPresent()) {
+            getGlobalRegionScheduler().get().execute(this, runnable);
+        } else {
+            super.runSync(runnable);
+        }
+    }
+
+    @Override
+    public int runAsyncRepeating(@NotNull Runnable runnable, long period) {
+        if (getGlobalRegionScheduler().isPresent()) {
+            getGlobalRegionScheduler().get().runAtFixedRate(this, scheduledTasks::add, 1, period);
+            return scheduledTasks.size() - 1;
+        } else {
+            return super.runAsyncRepeating(runnable, period);
+        }
+    }
+
+    @Override
+    public void runLater(@NotNull Runnable runnable, long delay) {
+        if (getGlobalRegionScheduler().isPresent()) {
+            getGlobalRegionScheduler().get().runDelayed(this, scheduledTask -> runnable.run(), delay);
+        } else {
+            super.runLater(runnable, delay);
+        }
+    }
+
+    @Override
+    public void cancelTask(int taskId) {
+        if (!folia) {
+            super.cancelTask(taskId);
+        }
+        scheduledTasks.get(taskId).cancel();
+    }
+
+    @Override
+    public void cancelAllTasks() {
+        if (!folia) {
+            super.cancelAllTasks();
+        }
+        getServer().getGlobalRegionScheduler().cancelTasks(this);
+    }
```

Comment:
This should still all be in it's own PaperTaskRunner class imo, even if you've simplified the logic